cli/test/test_scm.py

```diff
@@ -0,0 +1,278 @@
+        os.chdir('/tmp/test_folder/test_clone')
+        scm.clone(
+            'https://github.com',
+            'JayjeetAtGithub',
+            'github-actions-demo',
+            os.path.join(os.getcwd(), 'gad'),
+            'develop'
+        )
+        repo = git.Repo(os.path.join(os.getcwd(), 'gad'))
+        self.assertEqual(repo.active_branch.name, 'develop')
+        scm.clone(
+            'https://github.com',
+            'JayjeetAtGithub',
+            'github-actions-demo',
+            os.path.join(os.getcwd(), 'gad'),
+            'master'
+        )
+        repo = git.Repo(os.path.join(os.getcwd(), 'gad'))
+        self.assertEqual(repo.active_branch.name, 'master')
+
+    def test_parse(self):
+        test_url = "ssh://git@github.com:cplee/github-actions-demo.git"
+        self.assertRaises(SystemExit, scm.parse, test_url)
+        test_url = "https://github.com/cplee@master"
+        self.assertRaises(SystemExit, scm.parse, test_url)
+        test_url = "github.com"
+        self.assertRaises(SystemExit, scm.parse, test_url)
+        test_url = "@master"
+        self.assertRaises(SystemExit, scm.parse, test_url)
+        test_url = "http://gitlab.com/cplee/github-actions-demo.git"
+        parts = scm.parse(test_url)
+        self.assertTupleEqual(parts, (
+            'http://gitlab.com',
+            'gitlab.com',
+            'cplee',
+            'github-actions-demo',
+            '', None))
+        test_url = "https://github.com/cplee/github-actions-demo@master"
+        parts = scm.parse(test_url)
+        self.assertTupleEqual(parts, (
+            'https://github.com',
+            'github.com',
+            'cplee',
+            'github-actions-demo',
+            '',
+            'master'))
+        test_url = "github.com/cplee/github-actions-demo"
+        parts = scm.parse(test_url)
+        self.assertTupleEqual(parts, (
+            'https://github.com',
+            'github.com',
+            'cplee',
+            'github-actions-demo',
+            '',
+            None))
+        test_url = "cplee/github-actions-demo/path/to/action@develop"
+        parts = scm.parse(test_url)
+        self.assertTupleEqual(parts, (
+            'https://github.com',
+            'github.com',
+            'cplee',
+            'github-actions-demo',
+            'path/to/action',
+            'develop'))
+
+
+class TestScmWithoutGit(unittest.TestCase):
+    """Unit tests for popper.scm module in non git
+    environment."""
+
+    def setUp(self):
+        log.setLevel('CRITICAL')
+        if os.path.exists('/tmp/test_folder'):
+            shutil.rmtree('/tmp/test_folder')
+        os.makedirs('/tmp/test_folder')
+        os.chdir('/tmp/test_folder')
+        scm.clone(
+            'https://github.com',
+            'cplee',
+            'github-actions-demo',
+            os.path.join(os.getcwd(), 'github-actions-demo')
+        )
+        shutil.rmtree('/tmp/test_folder/github-actions-demo/.git')
```

Comment:
we can make this depend on `os.environ.get('POPPER_TEST_MODE')`. In this way we can have a single `TestScm` class

Reply:
Yeah, Done